src/components/tooltip/tooltip.ts

```diff
@@ -0,0 +1,192 @@
+import {Component, ComponentRef, Directive, Input, ElementRef, ViewContainerRef,
+    ChangeDetectorRef} from '@angular/core';
+import {Overlay} from '@angular2-material/core/overlay/overlay';
+import {OverlayState} from '@angular2-material/core/overlay/overlay-state';
+import {OverlayRef} from '@angular2-material/core/overlay/overlay-ref';
+import {ComponentPortal} from '@angular2-material/core/portal/portal';
+import {OverlayConnectionPosition, OriginConnectionPosition} from
+    '@angular2-material/core/overlay/position/connected-position';
+
+export type TooltipPosition = 'before' | 'after' | 'above' | 'below';
+
+@Directive({
+  selector: '[md-tooltip]',
+  host: {
+    '(mouseenter)': '_handleMouseEnter($event)',
+    '(mouseleave)': '_handleMouseLeave($event)',
+  }
+})
+export class MdTooltip {
+  visible: boolean = false;
+
+  /** Allows the user to define the position of the tooltip relative to the parent element */
+  private _position: TooltipPosition = 'below';
+  @Input('tooltip-position') get position(): TooltipPosition {
+    return this._position;
+  }
+  set position(value: TooltipPosition) {
+    if (value !== this._position) {
+      this._position = value;
+      this._createOverlay();
+      this._updatePosition();
+    }
+  }
+
+  /** The message to be displayed in the tooltip */
+  private _message: string;
+  @Input('md-tooltip') get message() {
```

Comment:
Leaving for now (currently, most of Material uses hyphens) - need to discuss a standard convention.